x/muxer/provider.go

```diff
@@ -0,0 +1,14 @@
+package muxer
+
+import (
+	"github.com/pulumi/pulumi/pkg/v3/codegen/schema"
+	"github.com/pulumi/pulumi/pkg/v3/resource/provider"
+	pulumirpc "github.com/pulumi/pulumi/sdk/v3/proto/go"
+)
+
+// Provider defines an interface which must be implemented by providers
+// that shall be used as mixins of a wrapped Terraform provider
+type Provider interface {
```

Comment:
This interface belongs in the main package perhaps, since it's part of defining ProviderInfo and sounds like we're committing to it, WDYT?

Reply:
@t0yv0 @iwahbe because the interface is only used when defining a mixin for a wrapped TF provider, I think it's reasonable to locate it in the `muxer` package. However if it shall be moved to the main package it must be renamed.

Reply:
@tmeckel I agree with @t0yv0 that this should be located in the `tfbridge` package. I have 2 reasons:
1. `x/muxer` is an implementation detail of the bridge, and shouldn't need to be referenced by external users.
2. I think interfaces should be defined where they are consumed or provided, and `x/muxer.Provider` is neither.

Reply:
@iwahbe @t0yv0 I think the interface should then be renamed like `MuxProvider`, `ProviderMux`, `MixinProvider`. Right?

Reply:
@iwahbe @t0yv0 I moved the intreface to the `pkg/tfbridge` package and renamed it to `ProviderMixin`, but that really stutters. We should find a common sense on mix, mux , muxing, mixing! 😁 

![image](https://github.com/pulumi/pulumi-terraform-bridge/assets/14177833/77f22c0f-ca1a-4fad-9373-51c62219989c)

Reply:
The field is called `MuxWith`, and under the hood it uses a `muxer`, so I think I would go with `mux` instead of `mixin`. Since I think this interface will only be used (at least externally to the bridge) with this field, I would call it `MuxWithProvider`.

Naming is hard ([`AugmentShimWithDisjointPF`](https://github.com/pulumi/pulumi-terraform-bridge/blob/ba7cc1d3083f782df65129423fa72e8cdf7eaac6/pf/internal/muxer/muxer.go#L64)), so I would love to hear @t0yv0's opinion on this.

Reply:
Naming is hard. The problem with "mixin" word is that in Pulumi this means something already like we have hand-written node code like https://github.com/pulumi/pulumi-aws/blob/master/sdk/nodejs/s3/s3Mixins.ts that mixes into the generated SDK. So that term is taken. 

The word "mux" is entering our lexicon from TF prior art (https://github.com/hashicorp/terraform-plugin-mux). between "mix" and "mux" I'd lean to "mux".

Reply:
@t0yv0 Okay, well then I'd propose `MuxWith []MuxProvider` or `MuxWith []ProviderMux`. I prefer the first alternative, because it makes clear for what purpose the interface has and when it'll be used then.

CC @iwahbe

Reply:
I also prefer `MuxWith []MuxProvider`. That sounds good to me.

Reply:
@iwahbe @t0yv0 renamed to `MuxProvider`